modules/users/server/controllers/users/users.authentication.server.controller.js

```diff
@@ -116,7 +116,7 @@ exports.oauthCallback = function (strategy) {
           return res.redirect('/authentication/signin');
         }
 
-        return res.redirect(redirectURL || sessionRedirectURL || '/');
```

Comment:
A PR would be fine, I think it would be a good idea to review the authentication process and rework the redirections.